### DISCUSSION TITLE
Ability to add multiple fallback routes

### DISCUSSION THREAD

**Discussion by indykoning** [laravel/framework#53307] [Ideas]
Laravel has a way to add [fallback routes](https://laravel.com/docs/11.x/routing#fallback-routes) to the router.
Which works great as a catch all, so code can determine wether the route matches something.
(e.g. a redirect table, dynamic routes/pages)

Now say you've installed something like [Statamic](https://github.com/statamic/cms) which works using a fallback route.
And you want to use a fallback route in your own project as well.
One of the two will never get fired, while the other causes a 404 to occur.

My idea is to use the [Route::isFallback](https://github.com/laravel/framework/blob/5527e72efe5d86cf83afe21944a47fe9a0bdc406/src/Illuminate/Routing/Route.php#L58) to try the next route in case a route with `isFallback === true` returns a 404.

This way you can still use a fallback route without losing the ability to use packages like Statamic.

If people agree i am willing to create a PR to enable this feature.

**Comment by shaedrich** [comment]
https://github.com/laravel/framework/blob/5527e72efe5d86cf83afe21944a47fe9a0bdc406/src/Illuminate/Routing/Router.php#L242-L249

`$placeholder` could be a random value instead of a fixed one 🤔

**Reply by indykoning** [reply]
It could be, unfortunately due to the catch all nature of this route (which makes sense) any further fallbacks will not be executed if this returns a 404.

I've already made a fallback controller with capability to run multiple fallback routes 
https://github.com/rapidez/core/pull/166/files
https://github.com/rapidez/core/pull/617/files

But this depends on packages allowing you to disable their fallback routes, and adding them as a fallback route to that controller.
So it's not an ideal solution.
Some light modification to Laravels router would make it possible out of the box
